utils/lib/Alert.ts

```diff
@@ -81,6 +138,76 @@ class Alert extends Component<QuickstartConfigAlert[], QuickstartAlertInput[]> {
       };
     });
   }
+
+  /**
+   * Static method that gets the alert policy associated with a quickstart and it's current data sources
+   * @returns - object with alert policy ids, required data sources and NGerrors
+   */
+  static async getAlertPolicyRequiredDataSources(quickstart: {
+    name: string;
+    dataSourceIds: string[];
+  }): Promise<
+    | { alertPolicy: AlertPolicyDataSource }
+    | { alertPolicy: null; errors: ErrorOrNerdGraphError[] }
+  > {
+    const { data, errors } = await fetchNRGraphqlResults<
+      AlertPolicyRequiredDataSourcesQueryVariables,
+      AlertPolicyRequiredDataSourcesQueryResults
+    >({
+      queryString: ALERT_POLICY_REQUIRED_DATA_SOURCES_QUERY,
+      variables: { query: `${quickstart.name} alert policy` },
+    });
+
+    const results = data?.actor?.nr1Catalog?.search?.results;
+
+    if (errors) {
+      return { alertPolicy: null, errors };
+    }
+
+    if (results === undefined || results.length === 0) {
+      const error = new Error(
+        `No alert policy for quickstart ${quickstart.name} exists`
+      );
+
+      return { alertPolicy: null, errors: [error] };
+    }
+
+    const alertPoliciesWithUpdatedDataSources = results?.map(
```

Comment:
At this point, I think we should be able to assume that `results` is defined and has a length.

```suggestion
    const alertPoliciesWithUpdatedDataSources = results.map(
```